src/main/java/net/sf/jabref/importer/fetcher/GoogleScholarFetcher.java

```diff
@@ -49,29 +40,84 @@
 
     private static final Log LOGGER = LogFactory.getLog(GoogleScholarFetcher.class);
 
-    private boolean hasRunConfig;
     private static final int MAX_ENTRIES_TO_LOAD = 50;
-    private static final String QUERY_MARKER = "___QUERY___";
     private static final String URL_START = "http://scholar.google.com";
     private static final String URL_SETTING = "http://scholar.google.com/scholar_settings";
     private static final String URL_SETPREFS = "http://scholar.google.com/scholar_setprefs";
-    private static final String SEARCH_URL = GoogleScholarFetcher.URL_START + "/scholar?q=" + GoogleScholarFetcher.QUERY_MARKER
+
+    private static final Pattern INPUT_PATTERN = Pattern.compile("<input type=([^ ]+) name=([^ ]+) value=([^> ]+)");
```

Comment:
put the pattern to the other patterns. Also you should have an order, so that final variables come first, then an empty line and then the patterns.